### DISCUSSION TITLE
Should Quartz 4.0 only support modern .NET runtimes (6.0+)?

### DISCUSSION THREAD

**Discussion by lahma** [quartznet/quartznet#2080] [Polls]
I'm beginning to question whether it makes sense to keep supporting full framework (,NET 4.8 and below) in v4 development. While it's still used a lot, the current 3.x is stable and feature rich and as such should cater for many needs. We can keep servicing 3.x with bug fixes and minimal improvements.

As resources are limited, it can be hard to justify trying to make things work both on Full Framework and the modern .NET 6.0. The plan would be to support at least the LTS release but probably keep support for longer time than needed.

**Comment by omrikaufman** [comment]
some products are hard to upgrade them to the .net core and should also enjoy the benefit of the new features in Quart .net 4

.net 4.8 support will be grate :)

**Comment by dinesh1011-cloud** [comment]
Hello Iahma
I m facing one issue can you please help me to resolve this defect 
In Simple trigger type when the job execution has completed than trigger get deleted from trigger table(QRTZ_TRIGGERS).
I have also created a defect. 
defect ID -: #2308

I have created the trigger with these two ways 
![image](https://github.com/quartznet/quartznet/assets/131234397/a2fd8327-ba2f-4464-a6d5-25065de57bf5)

![image](https://github.com/quartznet/quartznet/assets/131234397/c0be749b-8723-4043-8298-5832948a2fd1)

Thanks In advance

**Comment by A9G-Data-Droid** [comment]
I normally suggest supporting framework as long as Microsoft does. Currently there is net472 and standard2.0 where targeting only standard2.0 would allow compatibility with Framework. So I would start by dropping net472 and announcing that framework will be cut at some point. A deprecation warning might get more attention than this poll.

**Reply by lahma** [reply]
The 3.x line is quite capable of supporting full framework usage. Mostly about downstream consumers understanding that 3.x is just fine and we don't need to upgrade unless (if even then) we are going to target modern runtimes. 

We probably know for a fact that there's no big enterprise projects running on full framework relying on Quartz based on the sponsorships 😉

**Comment by Hawxy** [comment]
Projects stuck on .NET Framework are typically slow-moving and aren't going to rush to upgrade to Quartz 4.0 even if you did support it. Keeping v3 around as an LTS should be good enough for 99% of .NET Framework projects.

**Reply by madhub** [reply]
I agree,

**Comment by gitlsl** [comment]
https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core
By the time you release 4.0, the support for Runtime6.0 is almost over, so we will start with 8.0 directly？。。

**Reply by A9G-Data-Droid** [reply]
[November 12, 2024 - for reference](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core)

**Reply by lahma** [reply]
Yes, I'm leaning towards 8.0.

**Comment by lahma** [comment]
main (v4) now supports only NET 8 (and later).
